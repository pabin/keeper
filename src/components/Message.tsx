import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Icon from 'react-native-vector-icons/dist/FontAwesome';

import { colors } from '../styles/colors';

type MessageProps = {
  message: string;
};

const Message = ({ message }: MessageProps): JSX.Element => (
  <View style={styles.container}>
    <Icon name="info-circle" size={32} color={colors.blueGrey700} />
    <Text style={styles.textStyle}>{message}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },

  textStyle: {
    fontSize: 18,
    marginVertical: 12,
    color: colors.blueGrey700,
  },
});

export default Message;
