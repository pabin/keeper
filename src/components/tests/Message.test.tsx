import React from 'react';
import renderer from 'react-test-renderer';

import Message from '../Message';

test('Message renders correctly', () => {
  const tree = renderer.create(<Message message="This is message" />).toJSON();
  expect(tree).toMatchSnapshot();
});
