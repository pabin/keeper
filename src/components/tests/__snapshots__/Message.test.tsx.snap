// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#455A64",
          "fontSize": 32,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "color": "#455A64",
        "fontSize": 18,
        "marginVertical": 12,
      }
    }
  >
    This is message
  </Text>
</View>
`;

exports[`Spinner renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#455A64",
          "fontSize": 32,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "color": "#455A64",
        "fontSize": 18,
        "marginVertical": 12,
      }
    }
  />
</View>
`;
